plasmapy/particles/decorators.py

```diff
@@ -20,488 +18,803 @@
     InvalidParticleError,
     ParticleError,
 )
-from plasmapy.particles.particle_class import Particle
+from plasmapy.particles.particle_class import CustomParticle, Particle, ParticleLike
+from plasmapy.particles.particle_collections import ParticleList, ParticleListLike
+
+_basic_allowed_annotations = (
+    Particle,  # deprecated
```

Comment:
Good question!  One example would be functions like `atomic_number` which would not work for custom particles (but which should work for particle lists that contain real particles, but doesn't yet).

Reply:
So, in that case, do we want to deprecate `Particle` as an annotation?

Reply:
My main reason for wanting to deprecate `Particle` is that, because strings and ints are also allowed, `Particle` doesn't serve as a type hint annotation and will mess up static type checkers like [mypy](http://mypy-lang.org/).  In #1724, I've tentatively added [`RealParticleLike`](https://github.com/PlasmaPy/PlasmaPy/blob/70b479027345cc78505157c5e14455635172bc3c/plasmapy/utils/typing.py#L13) to possibly account for cases like this.  There might need to be a `RealParticleListLike` in there too.